consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/statemachine/IbftRoundFactory.java

```diff
@@ -110,7 +110,7 @@ public IbftRound createNewRoundWithState(
         roundState,
         blockCreator,
         protocolContext,
-        protocolSchedule.getByBlockNumber(roundIdentifier.getSequenceNumber()).getBlockImporter(),
+        protocolSchedule.getByBlockHeader(parentHeader).getBlockImporter(),
```

Comment:
This needs to get the block importer based on the block we are building represented by the sequence number, not the parent block.

Instead of passing in the block importer could just get the block importer from the protocol schedule in the IbftRound importBlockToChain method.

Reply:
Yeah, your right that the sequence number is the same as the chain height/block number. If a timestamp is used to schedule the Shanghai fork then there is no one-to-one mapping from timestamp to sequence/block number as there could be multiple rounds involved to create that block.

Why was suggesting to resolve the block importer from the protocol schedule in the IbftRound importBlockToChain is that happens once the block is finalised and we have a timestamp so at that point the ProtocolSchedule.getByBlockHeader could be used.